lib/optimizely/notification_center.rb

```diff
@@ -33,7 +33,7 @@ def initialize(logger, error_handler)
       @error_handler = error_handler
     end
 
-    def add_notification_listener(notification_type, notification_callback)
+    def add_notification_listener(notification_type, callback = nil, &block)
```

Comment:
this is wrong approach. 
either you should add two listeners method, not default argument.

Reply:
Ruby does not support method overloading.
We could only send in default arguments or like 'method_name(*args)'. 
https://www.ruby-forum.com/topic/890619
Proc and lambda are used as closures to send a block in arguments, which is already being used.